docs/docs/miscellaneous/hooks.mdx

```diff
@@ -0,0 +1,32 @@
+# Hooks
+
+Elf allows customizing some of the behavior via `elfHooks`
+
+## registerPreStoreUpdate
+
+Provide a function to modify the store value after reducers has run.
+
+The callback function takes three arguments:
+
+- `currentState` - state before reducers has run
+- `nextState` - state after reducers has run
+- `storeName` - the name of the store
+
+One scenario when this functionality could be useful is when a developer wants to prevent bugs associated with mutability by freezing the store value during the development:
+
+```ts
+import { elfHooks, deepFreeze } from '@ngneat/elf';
+
+if (!environment.production) {
+  elfHooks.registerPreStoreUpdate<typeof store>(
+    (currentState, nextState, storeName) => {
+      return deepFreeze(nextState);
+    }
+  );
+}
+
+const store = createStore(
```

Comment:
This will not be in the same file. Please be accurate to avoid future questions.

Reply:
technically, the hooks can be registered from any file, including this file. I can update docs to show that hooks are registered in a different file. Although I don't see a big difference one way or another.

Reply:
since we're removing the types from the hooks, we don't need to know the type of the store. I'll remove the whole `createStore` block from the hooks docs.